server/utils/premises/occupancy.ts

```diff
@@ -64,3 +71,39 @@ export const durationSelectOptions = (durationDays?: string): Array<SelectOption
     text: label,
     selected: value === durationDays || undefined,
   }))
+
+export const generateDaySummaryText = (
+  allCharacteristicAvailability: Array<Cas1PremiseCharacteristicAvailability>,
+): string => {
+  const overbookedCriteria: Array<Cas1SpaceBookingCharacteristic> = []
+  allCharacteristicAvailability.forEach((characteristicAvailability: Cas1PremiseCharacteristicAvailability) => {
+    const { characteristic, availableBedsCount, bookingsCount } = characteristicAvailability
+
+    if (bookingsCount > availableBedsCount) {
+      overbookedCriteria.push(characteristic)
+    }
+  })
+  return overbookedCriteria.length
+    ? `This AP is overbooked on spaces with the following ${overbookedCriteria.length > 1 ? 'criteria' : 'criterion'}: ${overbookedCriteria.map(characteristic => occupancyCriteriaMap[characteristic].toLowerCase()).join(', ')}`
+    : ''
+}
+
+const summaryRow = (key: string, value: string): SummaryListItem =>
+  value && {
+    key: textValue(key),
+    value: textValue(value),
+  }
```

Comment:
Could use the recent `summaryListItem` utility in `formUtils`? It doesn't check for `value` being truthy, but I am quite undecided as to whether the basic utility should 😄

Reply:
Makes sense.